### PR TITLE
DB queries calendars

### PR DESCRIPTION
**Pull request by HarshwardhanSingh**
This is for the database operations for **calendars controller**. But as we have refactored calendars controller and extracted API methods in different service class adding this will simply cause conflicts while merging.

So you can remove this as well.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -34,7 +46,6 @@
     t.string   "refresh_token"
     t.string   "email"
     t.string   "avatar"
-    t.boolean  "is_admin"
```
Comment (revathskumar) [inline db/schema.rb]: This shouldn't be a part of this PR.

File: db/schema.rb [hunk 2]
```diff
@@ -23,6 +23,18 @@
     t.integer  "user_id"
   end
 
+  create_table "leaves", force: :cascade do |t|
```
Comment (revathskumar) [inline db/schema.rb]: This shouldn't be a part of this PR.

File: db/schema.rb (hunk 1 above)
Comment (HarshwardhanSingh) [inline db/schema.rb]: Fixed it already but it's not showing the change.

Comment (revathskumar) [inline db/schema.rb]: Nope. I don't think this got fixed. Can you check again?

File: app/controllers/calendars_controller.rb [hunk 3]
```diff
@@ -47,31 +47,80 @@ def show
...
+  # Create calendar (only admin)
   def create
-    @calendar = {'summary' => params[:summary]}
-    service = @client.discovered_api('calendar' , 'v3')
-    @response = @client.execute(:api_method => service.calendars.insert,
-                  :body => JSON.dump(@calendar),
-                  :headers => {'Content-Type' => 'application/json'})
+    if(current_user.is_admin?)
+      @calendar = {'summary' => params[:summary]}
+      service = @client.discovered_api('calendar' , 'v3')
```
Comment (emilsoman) [inline app/controllers/calendars_controller.rb]: As @revathskumar mentioned, please move all the direct API calls into a custom class that abstracts away implementation details of the google calendar API. The controller should only be concerned about calling a method on a service object to create a calendar.

File: app/controllers/calendars_controller.rb [hunk 4]
```diff
@@ -47,31 +47,80 @@ def show
...
+      @response = @client.execute(:api_method => service.calendars.insert,
+                    :body => JSON.dump(@calendar),
+                    :headers => {'Content-Type' => 'application/json'})
+      @calendar_insert = Calendar.create(summary: @calendar["summary"],
+                          remote_id: JSON(@response.response.body)["id"],
+                          user_id: current_user.id)
```
Comment (emilsoman) [inline app/controllers/calendars_controller.rb]: Please create a "CalendarCreateService" that encapsulates the whole logic of creating a calendar.

File: app/controllers/calendars_controller.rb [hunk 5]
```diff
@@ -47,31 +47,80 @@ def show
...
+      redirect_to calendar_path(Base64.urlsafe_encode64(JSON(@response.response.body)["id"]))
+    else
+      redirect_to dashboard_path
+    end
```
Comment (emilsoman) [inline app/controllers/calendars_controller.rb]: Why are we using instance variables like `@response` and `@calendar` in this action ?

**Comment by deepak** [comment]
> will simply cause conflicts while merging.
> So you can remove this as well.

@HarshwardhanSingh no. let us not delete this PR

let us rebase this after the refactoring PRs are merged
and see:

1. if it adds any new features
2. comments by @emilsoman are fixed

**Comment by HarshwardhanSingh** [comment]
Yes all these comment are for extracting API methods in separate class which I did already in **refactoring_calendars_controller** and **refactoring_leaves_controller branch.**

**Comment by deepak** [comment]
closing PR as it has been fixed in another PR

**Pull request closed without merging**